LegendasTV: fix caching titles for wrong season

fernandog opened the pull request:
It should only cache titles that matches give season when episode type
@ratoaq2 if ok, I will commit upstream to the respective PR

ratoaq2 commented on medusa/subtitle_providers/legendastv.py:
```diff
@@ -258,6 +258,12 @@ def search_titles(self, title, season):
                     else:
                         logger.warning('No season detected for title %d (%s)', title_id, name)
 
+                # discard title mismatches on season
```
Now you have parts of the mismatches discard logic in `search_titles` and parts in `query`.

fernandog replied:
Title is associated to the season, so it must be here.

fernandog commented:
@ratoaq2 I rewrote. I had to rename `title` var because you override the one from argument when you `title = {'type': type_map[source['tipo']]}` - same with new param `year` I added and year from result

p0psicles commented:
@ratoaq2, as far as I can review it, it looks good. Shall we approve?
